frappe/core/doctype/user/user.py

```diff
@@ -205,7 +205,7 @@ def send_password_notification(self, new_password):
 						_update_password(user=self.name, pwd=new_password,
 							logout_all_sessions=self.logout_all_sessions)
 
-					if not self.flags.no_welcome_mail and self.send_welcome_email:
+					if not self.flags.no_welcome_mail and self.send_welcome_email==1:
```

Comment:
This does not make sense. The `== 1` in here is redundant, and it should look for if the value is set to 1 (or True) by default. The issue is that the value comes in a string, which you may have to set to int. I'd suggest trying this out instead:

```suggestion
					if not self.flags.no_welcome_mail and cint(self.send_welcome_email):
```